cmd/buildclient/main.go

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2022, Sylabs, Inc. All rights reserved.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+	"github.com/sylabs/scs-build-client/internal/app/buildclient"
+)
+
+const (
+	keyAccessToken      = "auth-token"
+	keySkipTLSVerify    = "skip-verify"
+	keyArch             = "arch"
+	keyArtifactFileName = "output"
+	keyFrontendURL      = "url"
+	keyForceOverwrite   = "force"
+	keyImageSpec        = "image-spec"
+)
+
+var Usage = func() {
+	fmt.Fprintf(os.Stderr, `
+Usage: buildclient [opts] <path to build definition>
+
+    Build and push artifact to cloud library:
+
+        buildclient --image-spec library://user/project/image:tag alpine.def
+
+    Build, push artfifact to cloud library and download locally:
+
+        buildclient --image-spec library://user/project/image:tag --output /tmp/alpine_3.sif alpine.def
+
+    Build and push ephemeral artifact to cloud library:
+
+        buildclient alpine.def
+
+    Note: ephemeral artifacts are short-lived and are usually deleted within 24 hours.
```

Comment:
I was initially confused by the `--image-spec` and `--output` options. This documentation helps, but I wonder if the usage can be simplified further.

I'd expect the three use cases are:

1. Build, and pull locally, because I have other work to do with the image (for example, sign it.)
1. Build, and push directly to the Library. If I don't need to work with the image further, there's no reason to retrieve it.
1. Build, but discard the resulting image. Useful for validating your definition file.

I imagine the first two will be more common, and so perhaps we could simplify the command format to use positional arguments for those? Something like:

1. `buildclient alpine.def file://output.sif`
1. `buildclient alpine.def library://user/project/image:tag`
1. `buildclient alpine.def`